### PR TITLE
[bug fixes - advanced sdxl lora script] - add left lora changes from canonical script and fix bug in snr gamma

### PR DESCRIPTION
**Pull request by linoytsaban**
This PR updates and fixes issues in the **advanced dreambooth lora sdxl training script**
1. updates the advanced script according to the **peft lora** changes made in the canonical script
2. fixes bug when using `min SNR gamma` and `prior preservation loss`: 
when using both, there is a shape mismatch between the instance `loss` and the `mse_loss_weights` when trying to perform the following operation: 
`loss = loss.mean(dim=list(range(1, len(loss.shape)))) * mse_loss_weights`
causing an error.
I think the reason is the shape of `timesteps` that is used to calculate the snr gamma - we use timesteps sampled for all images in the batch (i.e. both instance and class) rather than only instance.

### CODE REVIEW

**Comment by linoytsaban** [comment]
cc @apolinario @sayakpaul for review 🙏

**Comment by sayakpaul** [comment]
Should https://github.com/huggingface/diffusers/pull/6294 be closed?

**Comment by linoytsaban** [comment]
> Should #6294 be closed?
my bad!, This one should only be the snr gamma - prior preservation bug fix then, I'll close this.

Does the change for the snr gamma calculation make sense logically do you think?

**Comment by linoytsaban** [comment]
closing to not overlap with [#6294](https://github.com/huggingface/diffusers/pull/6294), will open a separate one for the snr gamma bug
